test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -564,6 +571,5 @@ void UcxConnection::error_handler(void *arg, ucp_ep_h ep,
     assert(ep == self->_ep);
     log() << "detected error " << ucs_status_string(status)
           << " on connection " << self << std::endl;
-    self->_status = status;
-    self->_context.disconnect(self);
+    self->_context.on_disconnect(self);
```

Comment:
better call user virtual function from main progress, by going through _closing_conns first

Reply:
why it's better?